### PR TITLE
Add golden tests for pretty printer

### PR DESCRIPTION
**Pull request by astynax**
- Add golden tests for pretty printer
- Actualize haddock for `pretty`
- Export forgotten (?) `(<|)` from `Toml.PrefixTree`

### CODE REVIEW

File: test/Test/Toml/Printer/Golden.hs [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+          , ("doo", mempty)
+          , ("baz", mempty)
+          ]
+    }
+
+test_prettyGolden :: TestTree
+test_prettyGolden =
+    testGroup "Toml.Printer"
+        [ test pretty "test/golden/pretty_default.golden"
+        , test (prettyOptions $ PrintOptions True 0)
```
[the remark this answers is not in the document]
Comment (astynax) [reply test/Test/Toml/Printer/Golden.hs]: Good idea!

File: src/Toml/PrefixTree.hs [hunk 2]
```diff
@@ -2,6 +2,7 @@
 
 module Toml.PrefixTree
        ( PrefixTree (..)
+       , (<|)
```
Comment (ChShersh) [inline src/Toml/PrefixTree.hs]: Thank you for reexporting this function! It was forgotten to reexport indeed.

File: test/Test/Toml/Printer/Golden.hs [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import Toml (AnyValue (..), TOML (..), Value (..))
+import Toml.PrefixTree ((<|))
+import Toml.Printer (PrintOptions (..), pretty, prettyOptions)
+
+import qualified Data.HashMap.Strict as HashMap
+import qualified Toml.PrefixTree as PrefixTree
+
+example :: TOML
+example = mempty
+    { tomlPairs  = HashMap.fromList
```
Comment (ChShersh) [inline test/Test/Toml/Printer/Golden.hs]: Actually, we have eDSL implemented by @vrom911 for creating TOML data types. It was designed specially for such cases. See the following module:

* https://github.com/kowainik/tomland/blob/master/src/Toml/Edsl.hs

And here is one usage example:

* https://github.com/kowainik/tomland/blob/3187fed8dc353b68553f229b5dfb76fdc4e5e9f0/examples/Playground.hs#L81-L95

So you can write this test in more convenient way. And we also will test our eDSL as well 🙂

Comment (astynax) [reply test/Test/Toml/Printer/Golden.hs]: :+1:

**Pull request merged**